src/lib.rs

```diff
@@ -333,12 +342,10 @@ impl_read_into!(i16, read_i16);
 impl_read_into!(i32, read_i32);
 impl_read_into!(i64, read_i64);
 
-// We can't cast to bool, so this requires a separate method.
+// This requires a separate method definition because read_bool()
+// doesn't take a number of bits argument.
 impl ReadInto for bool {
-    fn read(reader: &mut BitReader, bits: u8) -> Result<Self> {
-        match try!(reader.read_u8(bits)) {
-            0 => Ok(false),
-            _ => Ok(true),
-        }
+    fn read(reader: &mut BitReader, _: u8) -> Result<Self> {
```

Comment:
Oops, this is wrong! It won't advance the reader properly. I'll just drop this change.